campaigns/private-browsing/simple.html

```diff
@@ -0,0 +1,121 @@
+   }
+   #xbutton-{{ snippet_id }}:active {
+       background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAMAAAAp4XiDAAAAe1BMVEUAAAAAAAAAAAAAAAABAQEAAAABAQEAAAAAAAAAAAD9/f3Dw8MAAAAAAACRkZH6+vr5+fnv7+/i4uLd3d3Z2dmgoKBdXV0mJiYAAAAAAACPj4/t7e3t7e2ampoAAACamprOzs7MzMyYmJj///8QEBB5eXnw8PCWlpZAQEB2Ec74AAAAI3RSTlMABgseOio+GRA0+68iEn/39OXTzceOYUoWFX/h4Igjiby5hsu+k6AAAAGoSURBVEjHnZbpYoIwEISbZEMIl8oNam27eLz/E9YY22DdgmR+efAxGxJ2540SY/wmxog/ycvDVombVBvOY4ynKpFaQ3AVaC0TlXI2CXRCQtPHxWYdRetNEWcNSNFZiHYQMsjKCEeKyiyQwjiRhJKw3+GTdl8gFcWwUOg6R1J5rUXInokEqhX+o1UFiWXGxDvEOKEY3i0z9vjESX08+jAurMe0j+CO4UpXOKtKK/5rksp6NY+sapmyHxMBOb6gHAS/m3Ry734/nfFB55P7vJcdu5sEO0dchuOYOA4Xx2wDwe8r6cfEYBhHDGMms6thCkpXxmB0dISRK7UEZRCeNNHotpYhvhhFTWIqC01dBEMQiL0Mr3W1OkaCIQmMdcvMUgqkGIrAwiyGC9ggxVAEbkBwsytrJBiSwHVgkQgJhiQw8kG8CvNYvsdD9thKc2Cy1w9MJkOPY+l3+Fkqs4WvmNmZ7YIXeVm7OMiOLWxKxsSj9Xk0WI827jEsPEaSx+DzHq9uiAeHLTHED4Eb4lRU6P9Ghd5GhelAkhGBZDb2gI09QMQer3DlEeG+AWkO2NrlPO+kAAAAAElFTkSuQmCC");
+   }
+   #simple-snippet-{{ snippet_id }}:hover > #xbutton-{{ snippet_id }} {
+       visibility: visible;
+   }
+   #xbutton-{{ snippet_id }}:hover + #block-snippet-overlay-{{ snippet_id }} {
+       visibility: visible;
+       background-color: #eaeaea;
+       box-shadow: 0pt 0pt 0pt white;
+       border: solid 0pt white;
+   }
+ {% endif %}
+</style>
+
+<div class="snippet" id="simple-snippet-{{ snippet_id }}">
+  {% if blockable %}
+    <div class="block-snippet-button" id="xbutton-{{ snippet_id }}">
+    </div>
+  {% endif %}
+  {% if blockable or clickable %}
+    <div id="block-snippet-overlay-{{ snippet_id }}">
+    </div>
+  {% endif %}
+  <a href="{{ icon_url|safe }}" class="launch-pb">
+    <img class="icon" src="{{ icon }}" />
+  </a>
+  <p>{{ text|safe }}</p>
+</div>
+
+{% if clickable %}
+  <script type="text/javascript">
+   //<![CDATA[
+   (function() {
+       var snippet = document.getElementById('simple-snippet-{{ snippet_id }}');
+       snippet.addEventListener('show_snippet', function () {
+          var buttons = snippet.getElementsByClassName('launch-pb');
+
+          isTargetAvailable('privateWindow', function targetAvailable () {
+            sendMetric('impression-pb-available');
+            for (var i=0; i < buttons.length; i++) {
```

Comment:
Micro optimization for your loops that caches `buttons.length`:

`for (var i=0, k=buttons.length; i < k; i++) {`